cmake/stm32/utilities.cmake

```diff
@@ -12,18 +12,20 @@ function(stm32_util_create_family_targets FAMILY)
     endif()
 
     if(NOT (TARGET STM32::${FAMILY}${CORE_C}))
+        # Remove unused code, generate thumb code and enable all warnings
+        set(STM32_COMPILE_OPTIONS
+            -mthumb -Wall -ffunction-sections -fdata-sections
+        )
+
         add_library(STM32::${FAMILY}${CORE_C} INTERFACE IMPORTED)
         target_compile_options(STM32::${FAMILY}${CORE_C} INTERFACE 

```

Comment:
```suggestion
        # Remove unused code, generate thumb code and enable all warnings
        target_compile_options(STM32::${FAMILY}${CORE_C} INTERFACE 

```

Reply:
@atsju This comment will be wrong when we will update any of the flags. I see two fixes:
 1. Make the comment more basic "sets the compiler flags for the target blablabla"
 2.  List the current flags alongside a brief description of what each does.

I'm more for the solution 1 here.

Reply:
Updated

Reply:
Done